backend/src/main/java/pl/lodz/p/it/opinioncollector/userModule/user/UserManager.java

```diff
@@ -46,8 +46,39 @@ private Token generateAndSaveDeletionToken(User user) {
         return tokenRepository.save(deletionToken);
     }
 
+    private Token generateAndSavePasswordResetToken(String email) {
+        Token passwordResetToken = new Token();
+        User user = userRepository.findByEmail(email)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST));
+        passwordResetToken.setUser(user);
+        passwordResetToken.setToken(UUID.randomUUID().toString());
+        passwordResetToken.setType(TokenType.PASSWORD_RESET_TOKEN);
+        passwordResetToken.setCreatedAt(Instant.now());
+        return tokenRepository.save(passwordResetToken);
+    }
+
+    public void sendResetPassword(String email) {
+        try {
+            String resetPasswordToken = generateAndSavePasswordResetToken(email).getToken();
+            String link = "http://localhost:4200/resetConfirm/" + resetPasswordToken;
```

Comment:
Z environment brać base URL dla tego, żeby nie hardcode'ować
Ale to uwaga tylko, nie musicie naprawiać teraz